### DISCUSSION TITLE
how to make export data?

### DISCUSSION THREAD

**Discussion by Vasiliy-Makogon** [laravel/nova-issues#3690] [Q&A] [answered]
Hello. we use NOVA, standard means. We have a list, there are filters on this list. How can we unload data, for example, in Excel? 

I made an Action class, but as I understand it, a list of models is passed there. This will not be entirely correct if you have to download a large amount of data. How to do it right? 

```
class ProductsExportAction extends Action
{
    use InteractsWithQueue, Queueable;

    public $name = 'Excel export data';

    public function handle(ActionFields $fields, Collection $models)
    {
        print_r($models->count()); // 100 000 model objects??
    }

....
```

I suspect that you need to somehow get the query builder that generates a list of data for the view, its counterpart and call it in the ProductsExportAction::handle method. 

How to solve this problem correctly?

**Comment by crynobone** [comment, marked as answer]
See https://docs.laravel-excel.com/nova/1.x/
